core/service/transaction/delegate.ts

```diff
@@ -0,0 +1,128 @@
+import { ITransactionService } from 'core/service/transaction';
+import { IAssetDelegate, Transaction } from 'shared/model/transaction';
+import { Account } from 'shared/model/account';
+import { Delegate } from 'shared/model/delegate';
+import DelegateRepo from 'core/repository/delegate';
+import Response from 'shared/model/response';
+import AccountRepo from 'core/repository/account';
+import config from 'shared/util/config';
+import { ITableObject } from 'core/util/common';
+
+class TransactionDelegateService implements ITransactionService<IAssetDelegate> {
+
+    async create(trs: Transaction<IAssetDelegate>, data: IAssetDelegate): Promise<IAssetDelegate> {
+        const asset: IAssetDelegate = {
+            username: data.username,
+            url: data.url
+        };
+
+        if (asset.username) {
+            asset.username = asset.username.toLowerCase().trim();
+        }
+        return asset;
+    }
+
+    getBytes(trs: Transaction<IAssetDelegate>): Buffer {
+        return Buffer.from(trs.asset.username, 'utf8');
```

Comment:
> Buffer(null, 'utf8');
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type null
    at Function.from (buffer.js:195:11)
    at Buffer (buffer.js:170:17)

Reply:
Can better make the username required?

Reply:
It's required in model and is checked during verify, so the field is protected.